src/dbnode/storage/index/read_through_segment.go

```diff
@@ -140,6 +140,14 @@ func (r *ReadThroughSegment) ContainsField(field []byte) (bool, error) {
 	return r.segment.ContainsField(field)
 }
 
+// FreeMmap frees the mmapped data if any.
+func (r *ReadThroughSegment) FreeMmap() error {
+	if immSeg, ok := r.segment.(segment.ImmutableSegment); ok {
+		return immSeg.FreeMmap()
```

Comment:
Should we also perhaps free the mmap right after we checksum so that we don't get spikes at bootstrap time?